pyinstrument/renderers/profrenderer.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import marshal
+from typing import Any, Callable
+
+from pyinstrument import processors
+from pyinstrument.frame import Frame
+from pyinstrument.renderers.base import FrameRenderer, ProcessorList
+from pyinstrument.session import Session
+
+# pyright: strict
+
+
+class ProfRenderer(FrameRenderer):
+    """
+    Outputs a marshaled dict, containing processed frames.
+    """
+
+    output_file_extension = "prof"
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+
+    def render_frame(self, frame: Frame | None, stats: dict) -> dict:
+        if frame is None:
+            return {}
+
+        key = (frame.file_path or "", frame.line_no or 0, frame.function)
+        # Format is (call_time, number_calls, total_time, cumulative_time, callers)
+        val = [-1, -1, frame.time, frame.total_self_time, {}]
+        if frame.parent:
+            p = frame.parent
+            val[-1][(p.file_path or "", p.line_no or 0, p.function)] = val[:-1]
+        if key in stats:
+            return
```

Comment:
I think this is an issue - we shouldn't discard frames when they've already appeared in stats, we should combine their data. so times should accumulate. I think I have a fix for this, will push shortly.